docs/source/api/commands.rst

```diff
@@ -1,47 +1,626 @@
+    "Permission name", "name of granted permission", "permission is defined", "CanTransferAssets"
+
+
+Validation
+^^^^^^^^^^
+
+1. Account exists
+2. Transaction creator is allowed to grant this permission
+
+Remove signatory
+----------------
+
+Purpose
+^^^^^^^
+
+Purpose of remove signatory command is to remove a public key, associated with an identity, from an account
+
+Schema
+^^^^^^
+
+.. code-block:: proto
+
+    message RemoveSignatory {
+        string account_id = 1;
+        bytes public_key = 2;
+    }
+
+Structure
+^^^^^^^^^
+
+.. csv-table::
+    :header: "Field", "Description", "Constraint", "Example"
+    :widths: 15, 30, 20, 15
+
+    "Account ID", "id of account whom rights are granted", "already existent", "makoto@soramitsu"
+    "Public key", "Signatory to delete", "ed25519 public key", "407e57f50ca48969b08ba948171bb2435e035d82cec417e18e4a38f5fb113f83"
+
+Validation
+^^^^^^^^^^
+
+1. When signatory is deleted, we should check if invariant of **size(signatories) >= quorum** holds
+2. Signatory should have been previously added to the account
+
+Two cases:
+
+    Case 1. When transaction creator wants to remove signatory from their account and he or she has permission CanRemoveSignatory
+
+    Case 2. CanRemoveSignatory was granted to transaction creator
+
 Revoke permission
 -----------------
 
+Purpose
+^^^^^^^
+
+The purpose of revoke permission command is to revoke or dismiss given granted permission from another account in the network.
+
+Schema
+^^^^^^
+
+.. code-block:: proto
+
+    message RevokePermission {
+       string account_id = 1;
+       string permission_name = 2;
+    }
+
+Structure
+^^^^^^^^^
+
+.. csv-table::
+    :header: "Field", "Description", "Constraint", "Example"
+    :widths: 15, 30, 20, 15
+
+        "Account ID", "id of account whom rights are granted", "already existent", "makoto@soramitsu"
+        "Permission name", "name of granted permission", "permission was granted", "CanTransferAssets"
+
+Validation
+^^^^^^^^^^
+
+1. Transaction creator should have previously granted this permission to a target account
+
 Set account detail
 ------------------
 
+Purpose
+^^^^^^^
+
+Purpose of set account detail command is to set key-value information for a given account
+
+Schema
+^^^^^^
+
+.. code-block:: proto
+
+    message SetAccountDetail{
+        string account_id = 1;
+        string key = 2;
+        string value = 3;
+    }
+
+Structure
+^^^^^^^^^
+
+.. csv-table::
+    :header: "Field", "Description", "Constraint", "Example"
+    :widths: 15, 30, 20, 15
+
+    "Account ID", "id of account whom key-value information was set", "already existent", "makoto@soramitsu"
+    "Key", "key of information being set", "`[A-Za-z0-9_]{1,}`", "Name"
+    "Value", "value of corresponding key", "None", "Makoto"
+
+Validation
+^^^^^^^^^^
+
+Two cases:
+Case 1. When transaction creator wants to set account detail to his/her account and he or she has permission CanSetAccountInfo
```

Comment:
Please add empty lines to split the lines.
Please add indent for lines that start with "Case" word.